server/valgrind-out.txt

```diff
@@ -1,128 +1,141 @@
+--30091-- REDIR: 0x4913500 (libc.so.6:bcmp) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x491b5f0 (libc.so.6:memrchr) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4915090 (libc.so.6:strchrnul) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4912410 (libc.so.6:strcpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4913b90 (libc.so.6:strcasecmp_l) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4912720 (libc.so.6:strlen) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4912950 (libc.so.6:strncpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x49139d0 (libc.so.6:stpcpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4913470 (libc.so.6:memchr) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4912b80 (libc.so.6:strspn) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4913850 (libc.so.6:mempcpy) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4913b40 (libc.so.6:strncasecmp) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4915000 (libc.so.6:rawmemchr) redirected to 0x483f220 (_vgnU_ifunc_wrapper)
+--30091-- REDIR: 0x4a07790 (libc.so.6:__strrchr_avx2) redirected to 0x484e810 (rindex)
+--30091-- REDIR: 0x4a0b100 (libc.so.6:__memset_avx2_unaligned_erms) redirected to 0x4852770 (memset)
+--30091-- REDIR: 0x490f120 (libc.so.6:malloc) redirected to 0x4848820 (malloc)
+--30091-- REDIR: 0x4a07580 (libc.so.6:__strchrnul_avx2) redirected to 0x4853330 (strchrnul)
+--30091-- REDIR: 0x4a08e30 (libc.so.6:__strcpy_avx2) redirected to 0x484ee60 (strcpy)
+--30091-- REDIR: 0x49105a0 (libc.so.6:calloc) redirected to 0x484d9d0 (calloc)
+--30091-- REDIR: 0x4a07960 (libc.so.6:__strlen_avx2) redirected to 0x484ed10 (strlen)
+--30091-- REDIR: 0x490f460 (libc.so.6:free) redirected to 0x484b210 (free)
+--30091-- REDIR: 0x4a0a940 (libc.so.6:__memcpy_avx_unaligned_erms) redirected to 0x4852880 (memmove)
+--30091-- REDIR: 0x4913040 (libc.so.6:__GI_strstr) redirected to 0x4853ae0 (__strstr_sse2)
+--30091-- REDIR: 0x4a03c00 (libc.so.6:__memcmp_avx2_movbe) redirected to 0x4852010 (bcmp)
+--30091-- REDIR: 0x4a02ac0 (libc.so.6:__strcmp_avx2) redirected to 0x484fbd0 (strcmp)
+--30091-- REDIR: 0x4a0a900 (libc.so.6:__mempcpy_avx_unaligned_erms) redirected to 0x4853440 (mempcpy)
+--30091-- REDIR: 0x490f7c0 (libc.so.6:realloc) redirected to 0x484dc50 (realloc)
+==30091== 
+==30091== HEAP SUMMARY:
+==30091==     in use at exit: 80 bytes in 1 blocks
+==30091==   total heap usage: 958 allocs, 957 frees, 339,966 bytes allocated
+==30091== 
+==30091== Searching for pointers to 1 not-freed blocks
+==30091== Checked 108,208 bytes
+==30091== 
+==30091== 80 bytes in 1 blocks are still reachable in loss record 1 of 1
+==30091==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
+==30091==    by 0x109D5C: main (aesdsocket.c:293)
```

Comment:
@swapnilghonge this is pointing to a problem with your design, see my other comments about this.